groupByKey.ts

```diff
@@ -0,0 +1,19 @@
+type GroupsMap<T> = {
+  [key: string]: T[];
+};
+
+export function groupByKey<T>(items, key) {
```

Comment:
The `items` parameter should be typed as an array of objects, e.g., `items: T[]`, to ensure type safety. Similarly, the `key` parameter should be typed as a string, e.g., `key: string`.